srt-protocol/src/connection/mod.rs

```diff
@@ -355,27 +338,74 @@ impl DuplexConnection {
             // TODO: case UMSG_PEERERROR: // 1000 - An error has happened to the peer side
             PeerError(_) => todo!(),
             // TODO: case UMSG_EXT: // 0x7FFF - reserved and user defined messages
-            Srt(s) => self.handle_srt_control_packet(s),
+            Srt(s) => self.handle_srt_control_packet(now, s),
         }
     }
 
-    // handles a SRT control packet
-    fn handle_srt_control_packet(&mut self, pack: SrtControlPacket) {
+    fn handle_handshake_packet(&mut self, now: Instant, handshake: HandshakeControlInfo) {
+        if let Some(control) = self.handshake.handle_handshake(handshake) {
+            self.output.send_control(now, control);
+        }
+    }
+
+    fn handle_srt_control_packet(&mut self, now: Instant, pack: SrtControlPacket) {
         use self::SrtControlPacket::*;
         match pack {
-            HandshakeRequest(_) | HandshakeResponse(_) => {
-                warn!("Received handshake SRT packet, HSv5 expected");
-            }
+            HandshakeRequest(_) | HandshakeResponse(_) => self.warn("handshake", now, &pack),
             _ => unimplemented!(),
         }
     }
+
+    fn sender(&mut self) -> SenderContext {
+        SenderContext::new(
+            &mut self.status,
+            &mut self.timers,
+            &mut self.output,
+            &mut self.statistics,
+            &mut self.sender,
+        )
+    }
+
+    fn receiver(&mut self) -> ReceiverContext {
+        ReceiverContext::new(
+            &mut self.timers,
+            &mut self.output,
+            &mut self.statistics,
+            &mut self.receiver,
+        )
+    }
+
+    fn debug(&self, tag: &str, now: Instant, debug: &impl Debug) {
+        self.log(log::Level::Debug, tag, now, debug);
```

Comment:
Just a nitpick, but this should probably use the macro directly so you can compile various log levels out of the program

Reply:
Fair, good point. One goal to normalize the log formatting approach using structured logging of some kind.

Reply:
Fair enough--it's fine for now